### PR TITLE
feat(dm): using downstream DB to store lighting checkpoint

### LINKED ISSUE #8482: dm(cloud): meet bucket ratelimit when using gcp cloud storage as dumper storage
### What did you do?

run dm on gke

### What did you expect to see?

loader is normal

### What did you see instead?

loader meets:
```
[Lightning:Checkpoint:ErrUpdateCheckpoint]update checkpoint error: googleapi: Error 429: The rate of change requests to the object dbaas-nightly-us-west1-nightly-ms-73876-f332233f/dm-job-660216-tiflow-executor-0.tiflow-nightly-ms-73876-gke-us-west1-ca33bc82-88336618/6e370882-c471-42c0-b3a5-634ef3fe70de/3636303231362d736f757263652d373530303032/tidb_lightning_checkpoint.pb exceeds the rate limit. Please reduce the rate of create, update, and delete requests., rateLimitExceeded
```

### Versions of the cluster

DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):
```console
[086319d](https://github.com/pingcap/tiflow/commit/086319d24a77f5710005185fc53a53ec59cf0a08)
```
Upstream MySQL/MariaDB server version:
```console
5.7
```
Downstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):
```console
6.5
```
How did you deploy DM: tiup or manually?
```console
no
```
Other interesting information (system version, hardware config, etc):
```console
>
>
```

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

### PR DESCRIPTION
**Pull request by maxshuang**
### What problem does this PR solve?

Issue Number: close #8482  [tidb#42106](https://github.com/pingcap/tidb/issues/42106)

### What is changed and how it works?
Using downstream DB to store lightining checkpoint for cloud DM

#### Tests 

 - Unit test

### Release note 

```release-note
using downstream DB to store lighting checkpoint for cloud dm.
```

### CODE REVIEW

File: dm/loader/lightning.go [hunk 1]
```diff
@@ -325,14 +320,25 @@ func GetLightningConfig(globalCfg *lcfg.GlobalConfig, subtaskCfg *config.SubTask
...
-		return nil, err
+	if subtaskCfg.ExtStorage != nil {
+		// if we use bucket as dumper storage, write lightning checkpoint to downstream DB to avoid bucket ratelimit
+		// it's the default path for dm on cloud
+		cfg.Checkpoint.Driver = lcfg.CheckpointDriverMySQL
+		connParams, err := connParamFromDBConfig(&subtaskCfg.To)
+		if err != nil {
+			return nil, err
+		}
+		cfg.Checkpoint.MySQLParam = connParams
```
Comment (lance6716) [inline dm/loader/lightning.go:331]: I forget if we can left this field empty, and let lightning automatically fill the value from lightning's target database

**Review by lance6716 — APPROVED** [review]
need manually test

File: dm/loader/lightning.go (hunk 1 above)
Comment (maxshuang) [reply dm/loader/lightning.go:331]: I will check that logic.

Comment (maxshuang) [reply dm/loader/lightning.go:331]: image
Seems yes.

Comment (maxshuang) [reply dm/loader/lightning.go]: Already manully test.

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**